client/client.go

```diff
@@ -416,6 +416,21 @@ type HttpsClient struct {
 	Key         string
 }
 
+func (h *HttpsClient) Validate() {
+	if h == nil {
+		return
+	}
+	// Check if both the 'ServerCertificate' and 'PrivateKey' fields are set
+	if h.Certificate != "" || h.Key != "" {
+		if h.Certificate == "" {
+			log.Error("The 'PrivateKey' field is set in the mTLS configuration, but no 'ServerCertificate' is given. Both need to be present in order for mTLS to function")
+		}
+		if h.Key == "" {
+			log.Error("The 'ServerCertificate' field is set in the mTLS configuration, but no 'PrivateKey' is given. Both need to be present in order for mTLS to function")
+		}
+	}
```

Comment:
The error messages do not match what is actually being tested, and neither do the comments. I'm guessing this was started in one way, and then the code was later changed, but not the messages and comments. The code appears to be correct, so just changing the texts should be enough.

Reply:
It looks correct to me. 
It checks if only one of the two required fields are set
Case-Study:
1) If `Certificate` is set, and not `PrivateKey`
```
if h.Certificate != "" || h.key != "" {
```
is true, since `h.Certificate` is not `""`
```
if h.Certificate == "" {
```
is false, since, `h.Certificate` is set.
```
h.Key == "" {
```
is true, since `h.Key` is not set.
The following error message:
```
The 'ServerCertificate' field is set in the mTLS configuration, but no 'PrivateKey' is given. Both need to be present in order for mTLS to function
```
Then looks correct to me. Other case  follows by symmetry

It'd be fun if I was wrong here though, I had to stare at this for a bit :p

Reply:
@kacf

Reply:
The code is correct, as I said, it's the text which is wrong. The message

> The 'ServerCertificate' field is set in the mTLS configuration, but no 'PrivateKey' is given. Both need to be present in order for mTLS to function

is wrong because neither `ServerCertificate` nor `PrivateKey` are valid fields. They are called `Certificate` and `Key`. Same with the error messages, they are instructing the user to change fields that don't exist.

Reply:
Is changing it to `Certificate` and `Key` sufficient?

Reply:
Github is confused about mixing reviews and comments, apparently...

But yes, that should be good!